dbsnap_verify/rds_funcs.py

```diff
@@ -0,0 +1,172 @@
+from operator import itemgetter
+
+from random import choice
+
+from string import (
+    letters,
+    digits,
+)
+
+VALID_SNAPSHOT_TYPES = ["automated", "manual"]
+
+
+def generate_password(size = 9, pool = None):
+    """
+    Return a system generated password.
+    :param size:
+        The desired length of the password to generate. (Default 9)
+    :param pool:
+        Pool of chars to choose from. (Default digits and letters [upper/lower])
+    :returns: String (raw password)
+    """
+    if pool == None:
+        pool = letters + digits
+    return ''.join( [ choice( pool ) for i in range( size ) ] )
+
+
+def get_available_snapshots(session, db_id, snapshot_type=None):
+    """Returns DB snapshots in the available state for a given db id.
+    Args:
+        session (:class:`boto.rds2.layer1.RDSConnection`): The RDS api connection
+            where the database is located.
+        db_id (string): The database instance identifier whose snapshots you
```

Comment:
Should be `str` or `basestring`

Reply:
btw, I think these come from my dbsnap-copy docstrings, which were bad and need to be cleaned up :)